fendermint/vm/interpreter/src/fvm/checkpoint.rs

```diff
@@ -138,6 +139,38 @@ where
     }
 }
 
+/// Collect incomplete signatures from the ledger which this validator hasn't signed yet.
+///
+/// It doesn't check whether the validator should have signed it, that's done inside [broadcast_incomplete_signatures] at the moment.
+/// The goal is rather to avoid double signing for those who have already done it.
+pub fn unsigned_checkpoints<DB>(
+    gateway: &GatewayCaller<DB>,
+    state: &mut FvmExecState<DB>,
+    validator_key: PublicKey,
+) -> anyhow::Result<Vec<getter::BottomUpCheckpoint>>
+where
+    DB: Blockstore + Send + Sync + 'static,
+{
+    let mut unsigned_checkpoints = Vec::new();
+    let validator_addr = EthAddress::from(validator_key);
+
+    let incomplete_checkpoints = gateway
+        .incomplete_checkpoints(state)
+        .context("failed to fetch incomplete checkpoints")?;
+
+    for cp in incomplete_checkpoints {
+        let signatories = gateway
+            .checkpoint_signatories(state, cp.block_height)
```

Comment:
not for now, but a method `has_signed(state, height, addr)` can be more effective.